codegen/core/src/main/java/software/amazon/smithy/python/codegen/ClientGenerator.java

```diff
@@ -326,17 +326,26 @@ async def _handle_execution(
                                     )
                                 except SmithyRetryException:
                                     raise context_with_response.response
+                                logger.debug(
+                                    "Retry needed. Attempting request #%s in %s seconds.",
+                                    retry_token.retry_count + 1,
+                                    retry_token.retry_delay
+                                )
                                 await sleep(retry_token.retry_delay)
                                 current_body =  context_with_transport_request.transport_request.body
                                 if (seek := getattr(current_body, "seek", None)) is not None:
                                     await seek(0)
                             else:
                                 # Step 8: Invoke record_success
+                                logger.debug(
+                                    "Attempt %s succeeded. Not retrying request.",
```

Comment:
We should probably quantify what attempt succeeded, I don't know if there's enough context here to be immediately clear.

I'm also not sure if we need to log this at all as it may produce more noise than value. Generally we want to record exceptional cases rather than noting everything that happened.
```suggestion
                                    "Request attempt %s succeeded. Not retrying request.",
```

Reply:
Removed this in 7d90ac5ff71221af5b51649ea9b9a12adb739193

We can always add it back if we want to.